src/lib/functions/registry.mjs

```diff
@@ -181,6 +184,18 @@ export class FunctionsRegistry {
     return await listFunctions(...args)
   }
 
+  // Check if a function has a duplicate with a preceding extension
+  hasPrecedingDuplicate({ mainFile, name }) {
+    const registeredDupe = this.functions.get(name)
```

Comment:
Can we handle `undefined` here? I know that there is the `has` before calling this, but just to be sure, can we early exit if `registeredDupe` is undefined?

I tried to look at this function as it's own unit and right now it needs kind of a precondition (that there is a duplicate) that someone would need to be aware of when calling it.

```suggestion
    const registeredDupe = this.functions.get(name)
    if (!registeredDupe) {
        return false;
    }
```

Reply:
Good idea! ✅